src/AprsService.scala

```diff
@@ -263,13 +335,56 @@ class AprsService extends Service {
 	}
 	def sendPacket(packet : APRSPacket) { sendPacket(packet, "") }
 
+	def formatLocMice(symbol : String, status : String, location : Location) = {
+		val privambiguity = 5 - prefs.getStringInt("priv_ambiguity", 0)
+		val ambiguity = if (privambiguity == 5) 0 else privambiguity
+
+		Log.d("MICE", s"Set Ambiguity $ambiguity")
+		
+		val miceStatus = prefs.getString("p__location_mice_status", "Off Duty")
+		val (a, b, c) = AprsPacket.statusToBits(miceStatus)
+
+		val status_freq = AprsPacket.formatFreqMice(prefs.getStringFloat("frequency", 0.0f))
+		val (status_spd, course) = AprsPacket.formatCourseSpeedMice(location)
+
+		// Encoding process
+		val (infoString, west, longOffset) = AprsPacket.encodeInfo(location.getLongitude, status_spd, course, symbol)
+		val destString = AprsPacket.encodeDest(location.getLatitude, longOffset, west, a, b, c, ambiguity)
+		
+		val altitudeValue = if (prefs.getBoolean("priv_altitude", true)) {
+		  AprsPacket.formatAltitudeMice(location)
+		} else {
+		  None
+		}
+		val altString = altitudeValue.map(alt => AprsPacket.altitude(alt.toInt)).getOrElse("")
+
+		val formatPayload = infoString + 
+		  (if (altString.isEmpty) "" else altString) + 
+		  (if (status.isEmpty) "" else status) +
+		  (if (status.nonEmpty && status_freq.nonEmpty) " " else "") + 
+		  (if (status_freq.isEmpty) "" else status_freq) + "[1"
+
+		Log.d("formatLoc", s"MICE: $infoString $destString $altString")
+
+		val packet = newPacketMice(formatPayload, destString)
+
+		packet
+							
+	}
+
 	def postLocation(location : Location) {
 		var symbol = prefs.getString("symbol", "")
 		if (symbol.length != 2)
 			symbol = getString(R.string.default_symbol)
 		val status = prefs.getString("status", getString(R.string.default_status))
-		val packet = formatLoc(symbol, status, location)
```

Comment:
[important] I'd prefer to keep the logic in this method agnostic to the used compression mechanism, so my suggestion would be:

1. rename `formatLoc` to `formatLocUncompressed`
2. move the compressed location encoding to `formatLocCompressed`
3. make a new `formatLoc` wrapper method that will call and return the appropriate `formatLocX` according to the preference value.